File: packages/shared/src/types/dealers.ts
export interface DealerMetadataItem {
  dealerId: string;
  name: string | null;
  city: string | null;
  state: string | null;
  logoUrl: string | null;
}

export interface DealerMetadataResponse {
  dealers: DealerMetadataItem[];
  cities: string[];
}

export interface DealerRequestData {
  dealerName: string;
  dealerUrl: string;
  notes?: string;
}

File: backend/app/schemas/dealers.py
"""
Pydantic schemas for dealer metadata API endpoints
"""

from pydantic import BaseModel
from typing import List, Optional


class DealerMetadataItem(BaseModel):
    dealer_id: str
    name: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    logo_url: Optional[str] = None


class DealerMetadataResponse(BaseModel):
    dealers: List[DealerMetadataItem]
    cities: List[str]
